store/block_store.go

```diff
@@ -150,6 +156,27 @@ func (s *TendermintBlockStore) GetTxResult(txHash []byte) (*ctypes.ResultTx, err
 	}, nil
 }
 
+func (s *TendermintBlockStore) GetTxResultByHeightAndIndex(height *int64, index int) (*ctypes.ResultTx, error) {
+	blockResult, err := core.BlockResults(height)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(blockResult.Results.DeliverTx) <= index {
+		return nil, ErrIndexOutOfRange
+	}
+
+	return &ctypes.ResultTx{
+		Index:  uint32(index),
+		Height: *height,
+		TxResult: abci.ResponseDeliverTx{
+			Code: blockResult.Results.DeliverTx[index].Code,
+			Data: blockResult.Results.DeliverTx[index].Data,
+			Info: blockResult.Results.DeliverTx[index].Info,
```

Comment:
Remember the whole `Results.DeliverTx[index] == nil` incident?

Reply:
Check nil before dereferencing